### DISCUSSION TITLE
How to properly setup a dependency using a shared library?

### DISCUSSION THREAD

**Discussion by MGilleronFJ** [mesonbuild/meson#11684] [Q&A]
I wrote a simple project depending on SDL2:
```meson
project('Test', 'cpp', 'c', default_options: ['cpp_std=c++17'])

sdl2_dep = dependency('sdl2')

executable('TestBGFX',
    [
        'src/main.cpp'
    ],
    #win_subsystem: 'windows'
    dependencies: [
        sdl2_dep
    ]
)
```
I downloaded SDL2 from WrapDB as shown in the tutorial. The only difference is I didn't define `default_library=static`, as I would prefer to use a dynamic library.

The build succeeds, however the executable inside the build directory is unable to run because it can't find `SDL2.dll`. Is there something else I need to do for dynamic libraries to be placed next to the executable so I can run/debug the executable during development?

**Comment by eli-schwartz** [comment]
Running `meson devenv` inside the build directory will create a development shell which is set up with things like DLL paths to make things work for development.

(You'll still need to handle packaging in something like a .MSI installer, yourself.)

**Reply by MGilleronFJ** [reply]
That works for running via a console, but that's one more manual step... but then how do I debug with VSCode for example?

**Reply by eli-schwartz** [reply]
This is powered by devenv as well, see https://marketplace.visualstudio.com/items/mesonbuild.mesonbuild/changelog

The changelog entry for version 1.8.0 of the vscode extension describes how it works.

**Reply by MGilleronFJ** [reply]
Hmm having a ready-to-run build output would have allowed any debugger to work with it without having to use Meson through a plugin. I had an existing Meson build dir before I used VSCode, which I invoked from a command line, so I don't have `meson-vscode.env`.

In VSCode clicking the Meson icon leaves the left panel in an endless loading state. Is there a way I can generate the env file without the extension?

I tried restarting which prompted if I want the Meson extension to configure my project... which is weird because I already did `meson setup build` before (I prefer `build` to `builddir`). So I went to settings and changed the name of `builddir` folder to `build`, then clicked on `Yes` to configure the project. It took a while to do so (maybe it redownloaded SDL and rebuilt?) which ended up with still a `builddir` folder next to my already-setup `build` folder...

**Reply by eli-schwartz** [reply]
Yes, the extension should just be running `meson devenv --dump --dump-format vscode > meson-vscode.env`.

I have no experience with the extension myself since I'm a vim fan, I simply know that the vscode extension is popular with vscode users and it provides buttons that magically do what a lot of people like (and I guess magically do what a bunch of other people dislike). I know a lot more about the `meson` command interfaces.

That devenv "dump" command is most certainly available at any time after the build directory is configured, or even for whatever build directory name you like as long as you're happy to do whatever your development tooling wants to do in order to load that environment file in the place you outputted it to.
